irc/modes.go

```diff
@@ -261,6 +261,13 @@ func (channel *Channel) ProcessAccountToUmodeChange(client *Client, change modes
 		return l == r || umodeGreaterThan(l, r)
 	}
 
+	changed := false
+	defer func() {
```

Comment:
Can we move the `defer channel.stateMutex.Unlock()` line below (275) into here instead, so we're only calling one defer function and we can ensure it's going to be correct in future if the calling order of defer functions ends up changing?

Reply:
Ah, we use this sort of construction in other places in the codebase as-is. Something to keep an eye out in future, make sure nobody accidentally optimises it out (I could see myself not thinking enough and doing it, etc)